server/service/clustered_storage.go

```diff
@@ -285,86 +285,88 @@ func (d *clusterStorageProjectPushDriverInstance) runPeerSyncer(peerid uint64) {
+						if err != nil {
+							mylog.WithError(err).Info("Error renaming async retry queue")
+							return
+						}
 					}
+
+					return
 				}
 
-				return
+				// We have no object right now, but more might be inbound.
+				// Wait until we get notified the queues have changed, and then retry
+				mylog.Debug("Nothing to sync out, but not done. Wait")
+				d.objectSyncNewObjects.Wait()
+				d.objectSyncNewObjects.L.Unlock()
+				mylog.Debug("We were kicked")
+				continue
 			}
 
-			// We have no object right now, but more might be inbound.
-			// Wait until we get notified the queues have changed, and then retry
-			mylog.Debug("Nothing to sync out, but not done. Wait")
-			d.objectSyncNewObjects.Wait()
-			d.objectSyncNewObjects.L.Unlock()
-			mylog.Debug("We were kicked")
-			continue
-		}
-
-		mylog := mylog.WithField("objectid", nextentry)
-		mylog.Debug("Pushing single object to peer")
-
-		res := d.pushSingleObject(peerid, nextentry)
-		mylog.Debug("Results are in", "result", res)
-		if res == nil {
-			mylog.Debug("Object synced")
-		} else {
-			mylog.WithError(err).Info("Error during object sync")
-
-			if retryfile == nil {
-				if _, err := os.Stat(retrydir); os.IsNotExist(err) {
-					err := os.MkdirAll(retrydir, 0775)
+			mylog := mylog.WithField("objectid", nextentry)
+			mylog.Debug("Pushing single object to peer")
+
+			res := d.pushSingleObject(peerid, nextentry)
+			mylog.Debug("Results are in", "result", res)
+			if res == nil {
+				mylog.Debug("Object synced")
+			} else {
+				mylog.WithError(err).Info("Error during object sync")
+
+				if retryfile == nil {
+					if _, err := os.Stat(retrydir); os.IsNotExist(err) {
+						err := os.MkdirAll(retrydir, 0775)
+						if err != nil {
+							mylog.WithError(err).Info("Error creating retry dir")
+							break
+						}
+					}
+					var err error
+					retryfile, err = os.Create(retryfilename + ".inprogress")
 					if err != nil {
-						mylog.WithError(err).Info("Error creating retry dir")
-						break
+						mylog.WithError(err).Info("Error creating retry file")
+						retryfile = nil
+					} else {
+						fmt.Fprintf(retryfile, "project:%s\n", d.d.project)
 					}
 				}
-				var err error
-				retryfile, err = os.Create(retryfilename + ".inprogress")
-				if err != nil {
-					mylog.WithError(err).Info("Error creating retry file")
-					retryfile = nil
-				} else {
-					fmt.Fprintf(retryfile, "project:%s\n", d.d.project)
+				if retryfile != nil {
+					fmt.Fprintln(retryfile, nextentry)
 				}
-			}
-			if retryfile != nil {
-				fmt.Fprintln(retryfile, nextentry)
-			}
 
-			mylog.WithError(res).Info("Error syncing object to peer, queued for async")
-		}
+				mylog.WithError(res).Info("Error syncing object to peer, queued for async")
+			}
 
-		err = d.dbReportObject(nextentry, peerid, res == nil)
-		if err != nil {
-			d.errchan <- err
+			err = d.dbReportObject(nextentry, peerid, res == nil)
+			if err != nil {
+				d.errchan <- err
+			}
```

Comment:
This makes a function that had a fairly unweildly amount of indentation have even more indentation. Perhaps we should break the function up to make it easier to read?

Reply:
+1 I think breaking this into smaller functions would help.